src/bin/pg_autoctl/cli_create_drop_node.c

```diff
@@ -1214,7 +1214,7 @@ check_or_discover_hostname(KeeperConfig *config)
 									&monitorPort))
 		{
 			log_fatal("Failed to determine monitor hostname when parsing "
-					  "Postgres URI \"%s\"", config->monitor_pguri);
+					  "Postgres URI");
```

Comment:
I removed it here because I assumed that it was un-parsable. Do you think in the cases of un-parsable connection strings we should log it, even if it may reveal private information?

Reply:
If we failed to parse the monitor Postgres URI, there is nothing that can work in the deployment, and we want the user to fix that. How can they fix the situation without us showing what's wrong?

So while I have some level of sympathy to hiding private information in the connection string, my personal opinion is that allowing easy understanding of what's wrong in order to fix it takes priority. We don't know if the URI contains sensitive information: we failed to parse it. Let's at least show what we failed to parse, hopefully making it obvious what is to be fixed.

Reply:
Would a possible compromise be to print the mangled URI if and only if the logging level is `debug` or `verbose`? Most folks don't run in that configuration in production, and that's the place we feel that inapprioriately logging credentials is most sensitive. This way, if someone really needs to see what `pg_auto_failover` is tripping over, they can re-run in `verbose` mode, pick up the error, fix any mangling and also rotate credentials at that time?

I feel that this may be the right way to go, in that the biggest concern is credentials being silently printed to logs without the operator realizing. This way, we only print sensitive information to the log when we know that someone is explicitly choosing to look at the log.

Reply:
A good first step would be a version of this PR that omits the controversial points, and then we can open a new PR for what's left in your audit.

Reply:
I'll take a look and make the changes, I've been in the process of moving, so I haven't been very available lately. :)